bin/oneoff/backfill_pegasus_forms_sans_solr.rb

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env ruby
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'cdo/chat_client'
+require 'cdo/only_one'
+require 'retryable'
+require src_dir 'forms'
+require src_dir 'abort_form_error'
+
+# As the NULLness of various date columns indicates further processing is
+# necessary, we use this date to represent that an unrecoverable error happened.
+DATE_FOR_ERRORS = '1980-01-01 00:00:00'.freeze
+BATCH_SIZE = 15000
+
+# load helpers
+load pegasus_dir('helpers.rb')
+
+# ClassSubmission forms submitted via Pegasus up until the Solr deprecation have had additional
+# relevant metadata written to Solr by process_forms.  After the Solr deprecation,
+# new form submissions will have that additional metadata written to MySQL by
+# process_forms, going into the existing processed_data column.  This function
+# processes the earlier form submissions, generating that additional metadata
+# and storing it in the processed_data column.  It only works on forms that
+# have already been processed, and it also skips the update if the database
+# already contains the additional metadata, which means that it's idempotent.
+
+def process_existing_batch_of_forms
+  update_count = 0
+  already_written_count = 0
+  already_has_key_count = 0
+  error_count = 0
+
+  DB[:forms].where(kind: "ClassSubmission").limit(BATCH_SIZE).each do |form|
```

Comment:
I'm pretty sure this query as-is will only process the first `BATCH_SIZE` forms, not process in batches of `BATCH_SIZE` which is what I think you intend.

Take a look at other Sequel-based scripts in `/oneoff` for reference.

Reply:
Yes, this is why I bumped the batch size to 15,000 which should do all of the items in one attempt, though didn't explain well.  Does it seem reasonable to attempt that many modifications in one run?